### PR TITLE
fix(VSelect): correct chips spacing for compact and comfortable

### LINKED ISSUE #15686: [Bug Report][3.0.0-beta.9] `v-select` with `chips` props is not proper spacing
### Environment
**Vuetify Version:** 3.0.0-beta.9
**Vue Version:** 3.2.37
**Browsers:** Chrome 104.0.0.0
**OS:** Windows 10

### Steps to reproduce
Try to use `compact` and `comfortable` density with `chips` props.

### Expected Behavior
It should maintain proper spacing in `compact` and `comfortable` density.

### Actual Behavior
The chips touch the bottom line of `v-select`.

### Reproduction Link
<a href="https://codepen.io/Alokdev/pen/LYdopgq?editors=101" target="_blank">https://codepen.io/Alokdev/pen/LYdopgq?editors=101</a>

### PR DESCRIPTION
**Pull request by patrickclerc**
## Description
Adapt the SASS bottom margin of the select's chips for compact (3px) and comfortable density (2px) in order to avoid the chip and select underline collision. 

## Motivation and Context
Visually the chips in select field with density compact and comfortable are hoverred by the underline of the select which is not nice. 

fixes #15686

## How Has This Been Tested?

Visually tested using the below playground.

## Markup:
<details>

```vue
<template>
  <v-app>
    <v-container>
      <v-row>
        <v-col cols="4">
          <h2>Density : compact</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : comfortable</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : default</h2>
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            density="compact"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            density="comfortable"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            label="Standard variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="compact"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="comfortable"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            label="Plain variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="compact"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="comfortable"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            variant="outlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            variant="underlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="compact"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="comfortable"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
  </v-app>
</template>

<script>
  export default {
    name: 'Playground',
    data () {
      return {
        slider: 10,
        items: ['Foo', 'Bar', 'Fizz', 'Buzz'],
        selected: 'Foo',
      }
    },
  }
</script>
```
</details>

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VSelect/_variables.scss [hunk 1]
```diff
@@ -7,3 +7,4 @@ $select-line-height: 1.75 !default;
 $select-transition: .2s settings.$standard-easing !default;
 $select-chips-control-min-height: 64px !default;
 $select-chips-margin-top: 2px !default;
+$select-chips-margin-bottom: ('default': 0, 'comfortable': 0.5, 'compact': 0.75) !default;
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/_variables.scss:10]: density intervals should be whole numbers, `{ default: 0, comfortable: 1, compact: 2 }`.

File: packages/vuetify/src/components/VSelect/VSelect.sass [hunk 2]
```diff
@@ -4,6 +4,10 @@
 @use '../../styles/tools'
 @use './variables' as *
 
+@at-root
+  @include tools.density('v-input', $select-chips-margin-bottom) using ($modifier)
+    --select-chips-margin-bottom: #{0px + $modifier}
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/VSelect.sass:9]: I tested the following code with the updated densities and it appeared okay.

```sass
--select-chips-margin-bottom: #{0px + math.div($modifier, 2)}
```

**Review by johnleider — CHANGES_REQUESTED** [review]
VFileInput needs this updated as well. In addition, make sure you check VAutocomplete and VCombobox since they both use VSelect as a base composable.

**Comment by patrickclerc** [comment]
> VFileInput needs this updated as well. In addition, make sure you check VAutocomplete and VCombobox since they both use VSelect as a base composable.

Here is the new playground to check VSelect, VFileInput, VAutocomplete and VCombobox, one can note that the correction on VSelect doesn't apply to VAutocomplete and VCombobox. I need to adapt these two components also. 

`<template>
  <v-app>
    <v-container>
      <h1>VSelect</h1>
      <v-row>
        <v-col cols="4">
          <h2>Density : compact</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : comfortable</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : default</h2>
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            density="compact"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            density="comfortable"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            chips
            label="Standard variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="compact"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="comfortable"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            label="Plain variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="compact"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="comfortable"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            variant="outlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            variant="underlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="compact"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="comfortable"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-select
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
    <v-container>
      <h1>VCombobox</h1>
      <v-row>
        <v-col cols="4">
          <h2>Density : compact</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : comfortable</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : default</h2>
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            chips
            density="compact"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            chips
            density="comfortable"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            chips
            label="Standard variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="compact"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="comfortable"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            label="Plain variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="compact"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="comfortable"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            variant="outlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            variant="underlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="compact"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="comfortable"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-combobox
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
    <v-container>
      <h1>VAutocomplete</h1>
      <v-row>
        <v-col cols="4">
          <h2>Density : compact</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : comfortable</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : default</h2>
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            chips
            density="compact"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            chips
            density="comfortable"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            chips
            label="Standard variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="compact"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            density="comfortable"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            variant="plain"
            chips
            label="Plain variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="compact"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            density="comfortable"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Outlined variant"
            chips
            variant="outlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Underlined variant"
            chips
            variant="underlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="compact"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            density="comfortable"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-autocomplete
            :model-value="selected"
            :items="items"
            label="Solo variant"
            chips
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
    <v-container>
      <h1>VFileInput</h1>
      <v-row>
        <v-col cols="4">
          <h2>Density : compact</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : comfortable</h2>
        </v-col>
        <v-col cols="4">
          <h2>Density : default</h2>
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            chips
            density="compact"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            chips
            density="comfortable"
            label="Standard variant"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            chips
            label="Standard variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            variant="plain"
            chips
            density="compact"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            variant="plain"
            chips
            density="comfortable"
            label="Plain variant"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            variant="plain"
            chips
            label="Plain variant"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Outlined variant"
            chips
            density="compact"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Outlined variant"
            chips
            density="comfortable"
            variant="outlined"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Outlined variant"
            chips
            variant="outlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Underlined variant"
            chips
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Underlined variant"
            chips
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Underlined variant"
            chips
            variant="underlined"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Solo variant"
            chips
            density="compact"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Solo variant"
            chips
            density="comfortable"
            variant="solo"
          />
        </v-col>
        <v-col cols="4">
          <v-file-input
            :model-value="fileInput"
            label="Solo variant"
            chips
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
  </v-app>
</template>

<script>
  export default {
    name: 'Playground',
    data () {
      return {
        slider: 10,
        items: ['Foo', 'Bar', 'Fizz', 'Buzz'],
        selected: 'Foo',
        fileInput: [new File([''], 'filename.txt')],
      }
    },
  }
</script>
`

**Comment by patrickclerc** [comment]
Corrections done for VSelect, VAutocomplete, VCombobox and VFileInput. The results are shown in the below images : 

image
image
image
image

File: packages/vuetify/src/components/VAutocomplete/_variables.scss [hunk 3]
```diff
@@ -7,3 +7,4 @@ $autocomplete-line-height: 1.75 !default;
 $autocomplete-transition: .2s settings.$standard-easing !default;
 $autocomplete-chips-control-min-height: 64px !default;
 $autocomplete-chips-margin-top: 2px !default;
+$autocomplete-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact':2) !default;
```
Comment (johnleider) [inline packages/vuetify/src/components/VAutocomplete/_variables.scss:10]:
```suggestion
$autocomplete-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact': 2) !default;
```

File: packages/vuetify/src/components/VCombobox/_variables.scss [hunk 4]
```diff
@@ -7,3 +7,4 @@ $combobox-line-height: 1.75 !default;
 $combobox-transition: .2s settings.$standard-easing !default;
 $combobox-chips-control-min-height: 64px !default;
 $combobox-chips-margin-top: 2px !default;
+$combobox-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact':2) !default;
```
Comment (johnleider) [inline packages/vuetify/src/components/VCombobox/_variables.scss:10]:
```suggestion
$combobox-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact': 2) !default;
```

File: packages/vuetify/src/components/VFileInput/_variables.scss [hunk 5]
```diff
@@ -1,2 +1,4 @@
 // Defaults
 $file-input-chip-margin-inline-end: 4px !default;
+$file-input-chips-margin-top: 2px !default;
+$file-input-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact':2) !default;
```
Comment (johnleider) [inline packages/vuetify/src/components/VFileInput/_variables.scss:4]:
```suggestion
$file-input-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact': 2) !default;
```

File: packages/vuetify/src/components/VSelect/_variables.scss [hunk 6]
```diff
@@ -7,3 +7,4 @@ $select-line-height: 1.75 !default;
 $select-transition: .2s settings.$standard-easing !default;
 $select-chips-control-min-height: 64px !default;
 $select-chips-margin-top: 2px !default;
+$select-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact':2) !default;
```
Comment (johnleider) [inline packages/vuetify/src/components/VSelect/_variables.scss:10]:
```suggestion
$select-chips-margin-bottom: ('default': 0, 'comfortable': 1, 'compact': 2) !default;
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — APPROVED** [review]

**Pull request merged**